tasks/main.yml

```diff
@@ -2,6 +2,27 @@
 - name: Set version specific variables
   include_tasks: set_vars.yml
 
+- name: Scan sudoers
+  scan_sudoers:
+    output_parsed_configs: True
+    output_raw_configs: False
+  when: sudo_check_if_configured | bool
+
+- name: Compare
+  ansible.utils.fact_diff:
```

Comment:
I would prefer not to use a third party module - adding dependencies to external collections (ansible.utils is not built-in to ansible-core) is not a good idea unless the functionality is absolutely critical to the operation of the role.  In this case, it looks like fact_diff is mainly useful for giving user-friendly output.  I believe the objects `sudoers_files` and `sudo_sudoers_files` are both of type Python `dict` - you can just use `dict1 == dict2` to see if the dictionaries are equivalent.

If that doesn't work - then I suggest enhancing the scan_sudoers.py module to operate in a comparison mode e.g. if you give it a comparison dict argument, then it will compare the given dict with the contents of the sudoers file and output the differences, if any.

Reply:
I changed it for usual "equals" comparison and it seems to be working as expected.